types/node/globals.d.ts

```diff
@@ -320,4 +320,48 @@ declare namespace NodeJS {
     interface ReadOnlyDict<T> {
         readonly [key: string]: T | undefined;
     }
+
+    namespace undici {
+        type Request = typeof globalThis extends { onmessage: any } ? {} : import('undici-types').Request;
+        type Response = typeof globalThis extends { onmessage: any } ? {} : import('undici-types').Response;
+        type FormData = typeof globalThis extends { onmessage: any } ? {} : import('undici-types').FormData;
+        type Headers = typeof globalThis extends { onmessage: any } ? {} : import('undici-types').Headers;
+    }
+}
+
+declare function fetch(
+    input: import('undici-types').RequestInfo,
+    init?: import('undici-types').RequestInit,
```

Comment:
Do these need to be conditional? If you have both this and DOM, do you just get two signatures and everything's okay?

Reply:
The two overloads are almost unnoticeably different as it is, but that could be worth looking into. I’ll take a look.

Reply:
Yeah, it just occurred to me that no matter which overload is selected, they'll both return the same `Response`, so it can't matter, right?

Reply:
I took your suggestion and added a test that shows the minor way in which it matters.